### PR TITLE
[AppService]: fix #16838- az cli update app setting command always making slotsetting to true

### LINKED ISSUE #16838: WebApp:az cli update app setting command always making slotsetting to true
I am facing the same issue with az cli  2.16 version. Tried to upgrade it to 2.18 and still getting the same issue
az webapp config appsettings set -g dp-yamlcont-master -n dp-yamlcont-cont-test --slot staging  --settings @appSettings.json
The command above. always make the slot setting to true even if we define false in JSON

[

    {
        "name": "testNewApp" ,
        "value": "time",
        "slotsetting": "false"
    }
    ,
    {
        "name": "testNewApp1" ,
        "value": "time1",
        "slotsetting": "false"
    }
    ,
    {
        "name": "testNewApp2" ,
        "value": "time2",
        "slotsetting": "false"
    }
    ,
    {
        "name": "testNewApp3" ,
        "value": "time3",
        "slotsetting": "false"
    }
    ,
    {
        "name": "testNewApp2" ,
        "value": "@Microsoft.KeyVault(SecretUri=https://dpyamlcont.vault.azure.net/secrets/test1/f6d4d4f4dff74a689c29dfd180bc0740)",
        "slotsetting": "false"
    }
    ,
    {
        "name": "testNewApp3" ,
        "value": "@Microsoft.KeyVault(SecretUri=https://dpyamlcont.vault.azure.net/secrets/test2/2416121e38d74e7495f0d0fc2e3d9be6)",
        "slotsetting": "false"
    }
    ]

### PR DESCRIPTION
**Pull request by Kotasudhakarreddy**
**Description**

#16838
**Testing Guide**

az webapp config appsettings set -g dp-yamlcont-master -n dp-yamlcont-cont-test --slot staging --settings @appSettings.json

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by qwordy — APPROVED** [review]

**Comment by yonzhan** [comment]
AppService

**Comment by qwordy** [comment]
@panchagnula Can you help review it? Thanks

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -1080,7 +1080,7 @@
   - name: --settings
     short-summary: Space-separated appsettings in KEY=VALUE format. Use @{file} to load from a file.
   - name: --slot-settings
-    short-summary: Space-separated slot appsettings in KEY=VALUE format. Use @{file} to load from a file.
+    short-summary: Space-separated appsettings in KEY=VALUE format. Use @{file} to load from a file. Given setting are added to the configuration and marked as Deployment slot setting by default.
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:1083]: Is this supposed to be used only to mark app settings as Deployment slot settings? if so I would modify this as "
App settings that should be enabled as deployment slot settings. The app settings can be represented as 
space-separated  KEY=VALUE pair or Use @{file} to load from a file."

Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/_help.py:1083]: Did we test what happens if a user uses --settings with JSON file that has "slot settings: true" like portal client suggests? Does the command fail ?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -250,14 +250,16 @@ def update_app_settings(cmd, resource_group_name, name, settings=None, slot=None
...
     # pylint: disable=too-many-nested-blocks
     for src, dest in [(settings, result), (slot_settings, slot_result)]:
+        i += 1
         for s in src:
             try:
                 temp = shell_safe_json_parse(s)
                 if isinstance(temp, list):  # a bit messy, but we'd like accept the output of the "list" command
                     for t in temp:
-                        if t.get('slotSetting', True):
+                        if i != 1:
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:262]: I am not sure about this logic for using iterator inside for loop?

**Review by panchagnula — CHANGES_REQUESTED** [review]
Can you add comments to code explaining the logic for using variable for iteration & using for if check?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 2 above)
Comment (Kotasudhakarreddy) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Modified as per discussion

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py (hunk 1 above)
Comment (Kotasudhakarreddy) [reply src/azure-cli/azure/cli/command_modules/appservice/_help.py]: Modified the help text and tested as suggested. Found no issues.

**Comment by Kotasudhakarreddy** [comment]
@panchagnula please review

**Comment by Kotasudhakarreddy** [comment]
Ignore until `do-not-merge` tag is deleted.

**Comment by Kotasudhakarreddy** [comment]
Closing this PR as Rebase has not happened properly. Will create a new one.

**Pull request closed without merging**